pkgs/development/tools/rye/default.nix

```diff
@@ -53,6 +53,11 @@ rustPlatform.buildRustPackage rec {
 
   checkFlags = [
     "--skip=utils::test_is_inside_git_work_tree"
+
+    # The following require internet access to fetch a python binary
+    "--skip=test_lint_and_format"
+    "--skip=test_add_and_sync"
+    "--skip=test_empty_sync"
```

Comment:
I'm not familiar with `rye` so I've no idea if it's OK to disable those. These tests look like integration tests. I presume they only test the network by accident, and their intent must be to test some real/useful functionality.

Looking into this a bit deeper, I think we need to contact upstream and propose that they implement some kind of a dependency injection system that would allow a user to (optionally) disable the downloads and redirect them to use a system path instead. E.g. rye maybe could test for an environment variable like `RYE_USE_SYSTEM_PYTHON` being set and interpret its value as a path to the python binary to return instead of the downloaded artifact.

Then we could go whichever way: we could parse their 400-lines `download-urls.inc` (I forget the name) and extract the URLs of their prebuilt binaries (if we trust whoever indygreg on github is) in a FOD and use them during the tests, or we could even inject our own source-built python.

Reply:
Can we use `Custom Local Toolchains` for tests?
https://rye-up.com/guide/toolchains/#registering-toolchains

Reply:
That would be great indeed ! I tried to do it.
However, `rye` identifies the python releases based on the "sub-sub-version" (i.e. `3.12.2`).
We cannot be sure that our current `python3x` is actually the `python3.x.y` that it wants...

Reply:
Thanks for trying.
I hope to resolve this in the future, but it is not a blocker of the update.